### ISSUE TITLE
Potential incorrect indentation for logging in train_dreambooth.py

### ISSUE THREAD

**Issue by Men1scus** [huggingface/diffusers#12641]
**Description:**

In the `train_dreambooth.py` script, the logging and progress bar updates appear to be executed on every training step, even when using gradient accumulation. This might lead to incorrect or redundant logging.

The relevant code is located around lines 1393-1395:

https://github.com/huggingface/diffusers/blob/093cd3f040ee4f44908df8e1b441954f3f25c214/examples/dreambooth/train_dreambooth.py#L1346-L1395

The `global_step` is only incremented when `accelerator.sync_gradients` is true. However, the logging calls (`progress_bar.set_postfix` and `accelerator.log`) are outside this block. This means that when gradient accumulation is used, these lines are executed for every batch, but the `global_step` value passed to `accelerator.log` does not change until an optimization step occurs. This could result in multiple log entries for the same `global_step`.

It seems more appropriate to move the logging logic inside the `if accelerator.sync_gradients:` block to ensure that logging only happens once per optimization step.

**Proposed Change:**

```diff
  if accelerator.sync_gradients: 
      progress_bar.update(1) 
      global_step += 1 

      if accelerator.is_main_process: 
          # ... checkpointing and validation logic ...
 
- logs = {"loss": loss.detach().item(), "lr": lr_scheduler.get_last_lr()[0]} 
- progress_bar.set_postfix(**logs) 
- accelerator.log(logs, step=global_step)

+    logs = {"loss": loss.detach().item(), "lr": lr_scheduler.get_last_lr()[0]} 
+    progress_bar.set_postfix(**logs)
+    accelerator.log(logs, step=global_step)
```

Could you please confirm if this is the intended behavior or if the indentation should be corrected? Thank you!
